pkg/validations/cluster.go

```diff
@@ -27,15 +27,34 @@ func ValidateOSForRegistryMirror(clusterSpec *cluster.Spec, provider providers.P
 	}
 
 	machineConfigs := provider.MachineConfigs(clusterSpec)
-	if !cluster.Spec.RegistryMirrorConfiguration.InsecureSkipVerify || machineConfigs == nil {
+	if machineConfigs == nil {
 		return nil
 	}
 
 	for _, mc := range machineConfigs {
-		if mc.OSFamily() == v1alpha1.Bottlerocket {
+		if mc.OSFamily() == v1alpha1.Bottlerocket && cluster.Spec.RegistryMirrorConfiguration.InsecureSkipVerify {
 			return errors.New("InsecureSkipVerify is not supported for bottlerocket")
 		}
 	}
+
+	ociNamespaces := cluster.Spec.RegistryMirrorConfiguration.OCINamespaces
+	if len(ociNamespaces) == 0 {
+		return nil
+	}
+
+	ociNamespaceRegistryMap := make(map[string]bool)
+	for _, ns := range cluster.Spec.RegistryMirrorConfiguration.OCINamespaces {
+		ociNamespaceRegistryMap[ns.Registry] = true
+	}
+
+	for _, mc := range machineConfigs {
```

Comment:
So yes. This valid because it would only use `public.ecr.aws`, the other ones are just ignored. 
I also opened a docs PR that calls out that only public.ecr.aws works with bottlerocket.

https://github.com/aws/eks-anywhere/pull/7259/files

Reply:
but if other registry will be ignored, why don't we just fail the validation if its specified? as it won't do anything so we just error out telling user this won't work right now.

Reply:
Sure!  sounds good